File: apps/gateway/internal/slug/slug.go
// Package slug 实现站点 slug 的语法校验（不含保留词）。
// 规则见 packages/contracts/README.md；必须与 TS 实现通过同一组向量 vectors/slug.json。
package slug

import "strings"

const (
	MinLength = 3
	MaxLength = 32
)

// ValidSyntax 报告 s 是否符合 slug 语法。只接受 ASCII 小写字母、数字和连字符。
func ValidSyntax(s string) bool {
	if len(s) < MinLength || len(s) > MaxLength {
		return false
	}
	if s[0] == '-' || s[len(s)-1] == '-' || strings.Contains(s, "--") {
		return false
	}
	for i := 0; i < len(s); i++ {
		c := s[i]
		if (c < 'a' || c > 'z') && (c < '0' || c > '9') && c != '-' {
			return false
		}
	}
	return true
}

File: packages/contracts/src/slug.ts
export const SLUG_MIN_LENGTH = 3;
export const SLUG_MAX_LENGTH = 32;

/**
 * slug 语法校验（不含保留词）。规则见 packages/contracts/README.md，
 * Go 端实现必须通过同一组向量 vectors/slug.json。
 */
export function isValidSlugSyntax(input: string): boolean {
  if (input.length < SLUG_MIN_LENGTH || input.length > SLUG_MAX_LENGTH) return false;
  if (input.startsWith("-") || input.endsWith("-")) return false;
  if (input.includes("--")) return false;
  for (let i = 0; i < input.length; i++) {
    const c = input.charCodeAt(i);
    const isLower = c >= 0x61 && c <= 0x7a;
    const isDigit = c >= 0x30 && c <= 0x39;
    if (!isLower && !isDigit && c !== 0x2d) return false;
  }
  return true;
}
